File: app/cs50x2024/content/french/lectures_source_code/src6/1/positive.c
// Abstraction et domaine

#include <cs50.h>
#include <stdio.h>

int get_positive_int(void);

int main(void)
{
    int i = get_positive_int();
    printf("%i\n", i);
}

// Demander un entier positif à l'utilisateur
int get_positive_int(void)
{
    int n;
    do
    {
        n = get_int("Entier positif : ");
    }
    while (n < 1);
    return n;
}

File: app/cs50x2025/content/french/lectures_source_code/src6/1/positive.py
# Abstraction et portée

from cs50 import get_int


def main():
    i = get_positive_int()
    print(i)


# Demander à l'utilisateur un entier positif
def get_positive_int():
    while True:
        n = get_int("Entier positif : ")
        if n > 0:
            break
    return n


main()